src/main/_docs/admin-guide/admin-runbook.md

```diff
@@ -32,22 +33,206 @@ Zookeeper is a key-value store that is needed by Swarm in a clustered Codenvy se
 -A INPUT -m state --state NEW -m udp -p udp --dport 4789 -j ACCEPT
 ```
 
-**Step 2**: Create directories for Zookeeper:
+**Step 2**: Change the configuration of the Docker daemon to look like the below. On CentOS this file is found at `/etc/sysconfig/docker-network`:
 
 ```
-mkdir -p var/lib/zookeeper/datalog
+DOCKER_NETWORK_OPTIONS=' --bip=172.17.42.1/16 -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --cluster-store=zk://<%= scope.lookupvar('third_party::docker::install::docker_cluster_store') -%>:2181 --cluster-advertise=<%= scope.lookupvar('third_party::docker::install::docker_cluster_advertise') -%>:2375'
 ```
 
-**Step 3**: Start the zookeeper service:
+# Network Infrastructure
+We have 2 classes of instances:
+1. Master Nodes used for running internal Codenvy services.
+2. Workspace Nodes that runs user workspaces.
+
+We suggest separating traffic bewteen two subnets: one for Master and one for Machine Nodes. This keeps the traffic purpose and security properly segregated and allows the use of subnet masks, rather than individual IPs when writing firewall rules. VNET will use addresses 10.0.0.0/8 and two subnets 10.1.0.0/16 for Master and 10.2.0.0/16 for Machine Nodes.
+
+# Storage
+Cloud-based installs (AWS, Google Cloud, etc...) can quickly scale disk space up. To utilise this we suggest LVM and XFS and recommend using caching for both read and write.
+
+Machine Nodes typically require fast I/O access to give developers the best experience. This is espeically important with languages like node.js and PHP that can require access to a large number of small files. To optimize performance we suggest using SSD in all locations. 
+
+For Machine Nodes local storage is strongly preferred as it will provide the best performance.
+
+For the Master, LVM or RAID is recommended for redundancy and network attached storage can be used. If using Amazon Web Services we recommend LVM with snapshotting turned on for faster backups of the key data.
+
+## Directories
+
+For Master Node:
 
 ```
-docker run --name codenvy-zookeeper --restart always -d -p 2181:2181 -v /var/lib/zookeeper/data:/data -v /var/lib/zookeeper/datalog:/datalog zookeeper
+/var/log/journal
+/var/lib/pgsql
+/var/lib/docker
+/var/lib/docker-distribution
+/home/codenvy/codenvy-data/fs
+/home/codenvy/codenvy-data/che-machines-logs
+/home/codenvy/codenvy-data/logs
 ```
 
-**Step 4**: Change the configuration of the Docker daemon by editing `/etc/puppet/modules/third_party/templates/docker/docker-network.erb`:
+And for Machine Nodes:
 
 ```
-# /etc/sysconfig/docker-network
-DOCKER_NETWORK_OPTIONS=' --bip=172.17.42.1/16 -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --cluster-store=zk://<%= scope.lookupvar('third_party::docker::install::docker_cluster_store') -%>:2181 --cluster-advertise=<%= scope.lookupvar('third_party::docker::install::docker_cluster_advertise') -%>:2375'
+/var/log/journal
+/var/lib/docker
+/home/codenvy/codenvy-data
```

Comment:
remove line, we don't use synchronisation to workspace host, we do it directly to container.